test/functional/api/keymap_spec.lua

```diff
@@ -460,6 +460,65 @@ describe('nvim_get_keymap', function()
       desc = 'map description',
     }, api.nvim_get_keymap('n')[1])
   end)
+
+  it('can get abbreviations map', function()
+    exec_lua [[
+      local bufnr = vim.api.nvim_get_current_buf()
+      local GlobalCount = 0
+      vim.api.nvim_set_keymap('ia', 'foo', '', {
+        expr = true,
+        callback = function()
+          return tostring(GlobalCount)
+        end,
+      })
+
+      vim.api.nvim_buf_set_keymap(bufnr, 'ca', 'bar', '', {
+        expr = true,
+        callback = function()
+          return tostring(GlobalCount)
+        end,
+      })
+    ]]
```

Comment:
I think is better to just use an `:inoreabbr` and a `:cnoreabbr <buffer>` mapping here instead of two expression mappings, so that you won't need the `callback = nil` below and it is less verbose. Another test case for `!a` mode will also be good.